crates/project_panel/src/project_panel.rs

```diff
@@ -1307,22 +1315,32 @@ impl ProjectPanel {
             let mut tasks = Vec::new();
 
             for clipboard_entry in clipboard_entries.items() {
-                if clipboard_entry.worktree_id != worktree_id {
-                    return None;
-                }
                 let new_path =
                     self.create_paste_path(clipboard_entry, self.selected_entry_handle(cx)?, cx)?;
-                if clipboard_entries.is_cut() {
+                if clipboard_entry.worktree_id != worktree_id {
+                    let relative_path = self.create_relative_path(
+                        clipboard_entry,
+                        self.selected_entry_handle(cx)?,
+                        cx,
+                    )?;
                     self.project
                         .update(cx, |project, cx| {
-                            project.rename_entry(clipboard_entry.entry_id, new_path, cx)
+                            project.copy_relative_entry(entry.id, relative_path, new_path, cx)
                         })
                         .detach_and_log_err(cx);
                 } else {
-                    let task = self.project.update(cx, |project, cx| {
-                        project.copy_entry(clipboard_entry.entry_id, new_path, cx)
-                    });
-                    tasks.push(task);
+                    if clipboard_entries.is_cut() {
```

Comment:
I think we need to cut in case of the different workspace case too?
Looks like a good idea for another test.